pilot/pkg/proxy/envoy/v2/internalgen.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v2
+
+import (
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	"istio.io/istio/pilot/pkg/networking/util"
+
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	"github.com/golang/protobuf/ptypes/any"
+	"istio.io/istio/pilot/pkg/model"
+)
+
+const (
+	TypeURLConnections = "istio.io/connections"
+	TypeURLDisconnect  = "istio.io/disconnect"
+
+	// TODO: TypeURLReady - readiness events for endpoints, agent can propagate
+
+	// TypeURLNACK will receive messages of type DiscoveryRequest, containing
+	// the 'NACK' from envoy on rejected configs. Only ID is set in metadata.
+	// This includes all the info that envoy (client) provides.
+	TypeURLNACK = "istio.io/nack"
+
+)
+
+// InternalGen is a Generator for XDS status updates: connect, disconnect, nacks, acks
+type InternalGen struct {
+	Server *DiscoveryServer
```

Comment:
If we do the above comment and move the pushing logic out of here, we can avoid this circular dependency

Reply:
Right - the methods should probably by DiscoveryServer methods, on in InternalGen. 

Can we refactor in separate PR ?

Reply:
Sure